src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -392,7 +394,14 @@ private function dehydrateObjectValue(object $value, string $classType, ?string
             }
         }
 
-        throw new \LogicException(sprintf('Unable to dehydrate value of type "%s" for property "%s" on component "%s". Either (1) change this to a simpler value, (2) add the hydrateWith/dehydrateWith options to LiveProp or (3) set "useSerializerForHydration: true" on the LiveProp.', $value::class, $propertyPathForError, $componentClassForError));
+        $hydratedObject = [];
+        foreach ((new \ReflectionClass($classType))->getProperties() as $property) {
+            $propertyValue = $this->propertyAccessor->getValue($value, $property->getName());
+            $propMetadata = $this->liveComponentMetadataFactory->createLivePropMetadata($classType, $property->getName(), $property);
```

Comment:
This is cool! When a DTO is being (de)hydrated, it's effectively like each accessible property is given an automatic `#[LiveProp]`. That makes it easy to think about how all of this should work.